src/languages/java.rs

```diff
@@ -68,59 +94,53 @@ impl JavaProcessor {
         None
     }
 
-    /// Extracts the container (class or interface) with the method declaration and body.
-    ///
-    /// @example
-    ///
-    /// ```java
-    /// public class MyClass {
-    ///     public void method() {
-    ///         methodBody();
-    ///     }
-    /// }
-    /// ```
-    ///
-    /// is split into:
-    ///
-    /// ```java
-    /// public class MyClass {
-    ///     public void method() {
-    /// ```
-    /// and
-    ///
-    /// ```java
-    ///         methodBody();
-    ///     }
-    /// }
-    /// ```
-    fn extract_container_with_method(
+    /// Extracts the method with its full context from nested classes/interfaces
+    fn extract_method_with_context(
         &self,
-        content: &str,
-        container_node: Node,
         method_node: Node,
+        content: &str,
+        stack: &[(Node, String)],
     ) -> (String, String) {
-        // Find the container declaration start (including JavaDoc/comments before it)
-        let container_start = self.find_container_start_with_comments(container_node);
-        let container_body_start = self.find_container_body_start(container_node);
-
-        if let Some(body_start) = container_body_start {
-            let mut declaration = String::new();
-
-            // Add everything from the container start (including JavaDoc) up to the opening brace
-            let container_decl = &content[container_start..body_start];
-            declaration.push_str(container_decl);
+        let mut declaration = String::new();
+        
+        // Build the full context from the stack
+        for (i, (container_node, container_name)) in stack.iter().enumerate() {
+            if i > 0 {
+                declaration.push('\n');
+            }
+            
+            // Add container declaration with proper indentation
+            let container_start = self.find_container_start_with_comments(*container_node);
+            let container_body_start = self.find_container_body_start(*container_node);
+            
+            if let Some(body_start) = container_body_start {
+                let container_decl = &content[container_start..body_start + 1];
+                let indent = "    ".repeat(i);
+                let indented_decl = container_decl
+                    .lines()
+                    .map(|line| format!("{}{}", indent, line))
+                    .collect::<Vec<_>>()
+                    .join("\n");
+                declaration.push_str(&indented_decl);
+            }
+        }
+        
+        // Add the method declaration
+        let (method_declaration, method_body) = self.split_method_declaration_and_body(method_node, content);
+        let method_indent = "    ".repeat(stack.len());
+        let indented_method_decl = method_declaration
+            .lines()
+            .map(|line| format!("{}{}", method_indent, line))
+            .collect::<Vec<_>>()
+            .join("\n");
```

Comment:
Method indentation is just wrong. It should be keeping the indentation as in the source code. There is a start_position().column for that in tree sitter node